src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -135,6 +139,18 @@ export class TerminalInstance implements ITerminalInstance {
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
 
+		this._pidStack = [];
+		this._checkWindowShell = new Emitter<string>();
+		debounceEvent(this._checkWindowShell.event, (l, e) => e, 100, true)
```

Comment:
Maybe we should pull all this code out into another file so we don't clutter TerminalInstance? This would also simplify the function names as we can have `Windows` in the new class name.